core/network/src/protocol/sync.rs

```diff
@@ -441,6 +441,53 @@ impl<B: BlockT> ChainSync<B> {
 		})
 	}
 
+	/// Request syncing for the given block from given set of peers.
+	// This is similar to on_block_announce with unknown parent hash.
+	pub fn sync_fork(&mut self, peers: Vec<PeerId>, hash: &B::Hash, number: NumberFor<B>) ->
+		Vec<(PeerId, BlockRequest<B>)>
+	{
+		debug!(target: "sync", "Explicit sync request for block {:?} with {:?}", hash, peers);
+		if self.is_known(&hash) || self.is_already_downloading(&hash) {
+			debug!(target: "sync", "Refusing to sync known hash {:?}", hash);
+			return Vec::default();
+		}
+
+		let block_status = self.client.block_status(&BlockId::Number(number - One::one()))
+			.unwrap_or(BlockStatus::Unknown);
+		if block_status == BlockStatus::InChainPruned {
```

Comment:
As a follow-up, there should be a special mode where we just download the headers for this chain if requested, but we don't have the state to process them.

Reply:
https://github.com/paritytech/substrate/issues/3639 for a more detailed writeup of ^